specification/workloads/resource-manager/Microsoft.Workloads/common-types/v1/commonTypes.json

```diff
@@ -0,0 +1,69 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2022-10-01-preview",
+    "title": "Workloads Client",
+    "description": "Workloads client provides access to various workload operations"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {},
+  "definitions": {
+    "ManagedServiceIdentityType": {
+      "description": "Type of managed service identity (only None, UserAssigned types are allowed).",
+      "enum": [
+        "None",
+        "UserAssigned"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "ManagedServiceIdentityType",
+        "modelAsString": true
+      }
+    },
+    "UserAssignedServiceIdentity": {
+      "description": "Managed service identity (user assigned identities)",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Type of manage identity",
+          "$ref": "#/definitions/ManagedServiceIdentityType"
+        },
+        "userAssignedIdentities": {
+          "description": "User assigned identities dictionary",
+          "$ref": "../../../../../common-types/resource-management/v5/managedidentity.json#/definitions/UserAssignedIdentities"
+        }
+      },
+      "required": [
+        "type"
+      ]
+    }
+  },
+  "parameters": {}
```

Comment:
Why are these being redefined? Shouldn't they be referred directly from the common types? Should this file exist at all?

Reply:
This commonTypes file acts as a wrapper on top of actual [commonTypes ](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/common-types) since we have custom properties to be included for User MSI input. This file actually references the original commonTypes as done in Line 60

Reply:
The type you are trying to create here seems to be there already as ManagedServiceIdentity [https://github.com/Azure/azure-rest-api-specs/blob/077f1f8f0c8f7e5ac2ab17f9b80d36c56dc5156c/specification/common-types/resource-management/v5/managedidentity.json#L50](https://github.com/Azure/azure-rest-api-specs/blob/077f1f8f0c8f7e5ac2ab17f9b80d36c56dc5156c/specification/common-types/resource-management/v5/managedidentity.json#L50). It seems you are removing fields instead of adding.

Reply:
For our service in ManagedServiceIdentityType, only None, UserAssigned types are allowed whereas in actual commonTypes both SystemAssigned and UserAssigned types are allowed. For this purpose, we have defined our own ManagedServiceIdentityType definition. Since only UserAssigned type is allowed in our service we have created definition for that as well and inside it we refer to actual commonTypes file for the common properties to reuse instead of redefining those in our swagger